src/main/java/com/hendzior/veterinary/model/Animal.java

```diff
@@ -0,0 +1,66 @@
+package com.hendzior.veterinary.model;
+
+import javax.persistence.*;
+import java.time.Year;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+
+@Entity
+@Inheritance(strategy = InheritanceType.JOINED)
+public class Animal {
+
+    @Transient
+    private static final AtomicLong count = new AtomicLong(0);
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    protected Long id;
+    protected String name;
+    protected String gender;
+    protected String type;
+    protected int age;
+    @ManyToOne
+    @JoinColumn(name = "CUSTOMER_ID")
+    protected Customer customer;
+
+
+    public Animal() {
+    }
+
+     public Animal(String name, String gender, int age, String type, Customer customer) {
+        this.name = name;
+        this.gender = gender;
+        this.age = (Year.now().getValue() - age);
+        this.type = type;
+        this.customer = customer;
+        this.id = count.incrementAndGet();
```

Comment:
Wow. What's that? I thought hibernate or db will handle ids by itself

Reply:
Of course but when app is working with local data, the id wont generate by itself. 
What is a better way to generate id's?

Reply:
But I'm dinosaur.. ;)

Reply:
We used GenerationType.SEQUENCE  but as I said, it was veeery long time ago.
Your solution is ok for me as long as it's working :)